server/bin/pbench-audit-server.sh

```diff
@@ -713,10 +715,10 @@ function verify_archive {
     return $cnt
 }
 
-log_init $PROG
-
-# Initialize index mail content
-> ${index_content}
+# Ensure we have an existing final report file.
+report=${LOGSDIR}/${PROG}/report.latest.txt
+report_prev=${LOGSDIR}/${PROG}/report.prev.txt
+mv ${report} ${report_prev} 2>/dev/null
 > ${report}
```

Comment:
You're sure that you want to create the `${report}` file unconditionally?...you don't want to wait for the first `>>` operator to do that (in the hopes that maybe it will never be created at all)?

And, springboarding from @npalaska's question, are you sure that we want to do the `mv` at line 721 if `${report}` is empty?  There's a `[[ -s ${report} ]]` test which we could use to skip it.

Reply:
I have updated the code so that only if the intermediate report makes it all the way through without any errors will it replace the "latest" report.

The previous "latest" report is always saved.

So if a new "latest" report fails to be created, the previous report will linger until a new one comes along.

Reply:
`${report}` is cleaned up along with the rest of the workdir, so creating it empty sounds OK to me, although waiting is OK as well (it's rather unlikely that it will exist already).

As for checking for empty, does the absence of the report file mean the same thing as the presence of an empty  report file? Again, I don't think it makes much difference, as long as one does not depend on the absence of the file _only_ to declare "no problem".